lib/collection/property-list.js

```diff
@@ -272,6 +272,42 @@ _.extend(PropertyList.prototype, /** @lends PropertyList.prototype */ {
             prev = parent;
             parent = parent.__parent;
         }
+    },
+
+    toJSON: function () {
+        if (!this.count()) {
+            return;
+        }
+
+        return _.map(this.members, function (member) {
+            return _.reduce(member, function (accumulator, value, key) {
```

Comment:
This logic is common to the PropertyList and PropertyBase. Putting it in `util.js` is not an option because of cyclic dependency issues.. will figure out a better place for this. 

Suggestions welcome :)